src/light_client.rs

```diff
@@ -57,7 +58,13 @@ pub async fn run(
 					}
 					let commitment = header.extrinsics_root.commitment.clone();
 
-					let positions = rpc::generate_random_cells(max_rows, max_cols);
+					let cell_count: u32;
+					if confidence == 100f64 {
+						cell_count = -((1f64 - 99.9f64 / 100f64).log2()).ceil() as u32;
+					} else {
+						cell_count = (-((1f64 - confidence / 100f64).log2())).ceil() as u32;
```

Comment:
Can we write test cases to ascertain correctness? For example, giving `confidence=60` may give output `1` whereas `1` sample can only give atmost `50%` confidence.

Reply:
yeh sure! But how its being checked is if we put 60 as confidence then it checks for number on cells which gives >=60.

Reply:
I don't think `cell_count` of `1` gives `>=60` confidence.